program.py

```diff
@@ -181,990 +160,207 @@ def send_omf_message_to_endpoint(message_type, message_omf_json, action='create'
+        with open(
+            f"{app_path}/{filename}",
+            "r",
+        ) as f:
+            loaded_json = json.load(f)
+    except Exception as error:
+        print(f"Error: {str(error)}")
+        print(f"Could not open/read file: {filename}")
+        exit()
 
-# ************************************************************************
-# Helper functions: REQUIRED: create a JSON message that contains data values
-# for all defined containerids
-#
-# Note: if you do not send one of the values for the container, Relay
-# will emit the default value for missing property - it is the default
-# behavior of JSON serialization; this might lead to undesireable
-# results: for example, putting a value of zero into referenced PI
-# point
-# ************************************************************************
+    return loaded_json
 
 
-def getConfig(section, field):
-    # Reads the config file for the field specified
-    config = configparser.ConfigParser()
-    config.read('config.ini')
-    return config.has_option(section, field) and config.get(section, field) or ""
+def get_config():
+    ''' Return the config.json as a config file, while also populating base_endpoint, omf_endpoint, and default values'''
+    global endpoint_types
 
-# ************************************************************************
-# Note: PI points will be created on the first data value message
-# arrived for a given container
-# ************************************************************************
+    # Try to open the configuration file
+    endpoints = get_json_file("config.json")["endpoints"]
+
+    # for each endpoint construct the check base and OMF endpoint and populate default values
+    for endpoint in endpoints:
+        endpoint_type = endpoint["endpoint-type"]
+
+        # If the endpoint is OCS
+        if endpoint_type == endpoint_types[0]:
+            base_endpoint = f"{endpoint['resource']}/api/{endpoint['api-version']}" + \
+                f"/tenants/{endpoint['tenant']}/namespaces/{endpoint['namespace']}"
+
+        # If the endpoint is EDS
+        elif endpoint_type == endpoint_types[1]:
+            base_endpoint = f"{endpoint['resource']}/api/{endpoint['api-version']}" + \
+                f"/tenants/default/namespaces/default"
+
+        # If the endpoint is PI
+        elif endpoint_type == endpoint_types[2]:
+            base_endpoint = endpoint["resource"]
+
+        omf_endpoint = f"{base_endpoint}/omf"
+
+        # add the base_endpoint and omf_endpoint to the endpoint configuration
+        endpoint["base-endpoint"] = base_endpoint
+        endpoint["omf-endpoint"] = omf_endpoint
+
+        # check for optional/nullable parameters
+        if "verify-ssl" not in endpoint or endpoint["verify-ssl"] == None:
+            endpoint["verify-ssl"] = True
 
+        if "use-compression" not in endpoint or endpoint["use-compression"] == None:
+            endpoint["use-compression"] = False
```

Comment:
i would typically say default should be compression, idk, i could be convinced otherwise